src/rust/engine/process_execution/remote/src/remote_cache/reapi.rs

```diff
@@ -29,13 +29,25 @@ impl Provider {
       instance_name,
       action_cache_address,
       root_ca_certs,
+      mtls_certs,
+      mtls_key,
       headers,
       concurrency_limit,
       rpc_timeout,
     }: RemoteCacheProviderOptions,
   ) -> Result<Self, String> {
-    let tls_client_config = if action_cache_address.starts_with("https://") {
-      Some(grpc_util::tls::Config::new_without_mtls(root_ca_certs).try_into()?)
+    let needs_tls =
+      action_cache_address.starts_with("https://") || action_cache_address.starts_with("grpcs://");
```

Comment:
`grpcs://` check is unnecessary. See earlier comment about how `grpcs://` is translated to `https://`.